gui/qt/history_list.py

```diff
@@ -187,8 +188,22 @@ def update_labels(self):
         for i in range(child_count):
             item = root.child(i)
             txid = item.data(0, Qt.UserRole)
-            label = self.wallet.get_label(txid)
-            item.setText(3, label)
+            h_label = self.wallet.get_label(txid)
+            current_label = item.data(3, Qt.EditRole)
```

Comment:
`current_label = item.text(3)`  (not idem.data)  Column 3 will always return `None` for `item.data(3)`. Column 0 is the only thing that has item.data() there.  (FYI: `.data` is for "hidden" metadata you can attach to columns whereas `.text()` is for real UI visible stuff)

Reply:
Actually item.data(3, Qt.EditRole) does return the same string as item.text(3) but I see your point. Fixed in latest commit.

Reply:
Oh it does? I didn't know that -- I guess if nothing was set it falls back to the text, eh?  Strange.. the Qt docs don't talk about that at all ... ok thanks for changing it.